submariner/multiple-active-gateways.md

```diff
@@ -0,0 +1,660 @@
+is to centrally allocate IPs and write that information to the submariner globalnet specific Kubernetes objects which are
+then processed by Gateway controllers, not manipulate anything in the datapath.
+
+### Gateway Datapath Changes
+
+As mentioned above multiple Globalnet controllers were mirrored to run in the Gateway process in order to program all of
+the datapath rules (Iptables, ipset etc) needed for complete functionality of the globalnet feature. Specifically, these
+globalnet specific datapath controllers serve to watch submariner globalnet K8s objects (such as the
+`clusterglobalingressip` and `globalingressip` objects) and write datapath rules based on the data embedded in those objects.
+It then updates the status of those objects to let the users know the datapath rules have been properly implemented.
+
+Many of the actual iptables rules remained the same, however a few changes needed to occur.  Firstly the globalnet GW SNAT
+range for each GW within a cluster is now meant to be unique and align with the `clusterglobalingressIP` object.
+
+```shell
+-A SM-GN-EGRESS-CLUSTER -s <Local Pod CIDR> -m mark --mark 0xc0000/0xc0000 -j SNAT --to-source <IP Range Defined in the
+GW’s clusterGlobalEgressIPObject>
+```
+
+The other major change arose from the need to ensure that traffic within a single connection was always routed back to
+the same local GW,  maintaining NAT state. To accomplish this, routes in table 100 for every IP in the GWs SNAT range are
+added to each remote GW as follows.
+
+```shell
+Routing Table 100:
+<Remote GW 1 SNAT IP 1> via <Remote GW 1 IP> dev vxlan-tunnel metric 98
+…
+<Remote GW 1 SNAT IP X> via <Remote GW 1 IP> dev vxlan-tunnel metric 98
+<Remote GW 2 SNAT IP 1> via <Remote GW 2 IP> dev vxlan-tunnel metric 98
+…
+<Remote GW 2 SNAT IP X> via <Remote GW 2 IP> dev vxlan-tunnel metric 98
+```
+
+## Scaling considerations
+
+* Additional tunnels between clusters are required for this design.
+
+## Notes and Assumptions
+
+* The initial POC will use standard external IP addresses for gateways, but load balancer mode is
+  a high priority for the final implementation.
+* The initial design and POC is focused on CNIs that use the kubeproxy route agent handler that establishes VXLAN
+  tunnels between worker nodes and gateways. Other options need to be investigated and supported.
+* The multipath routes require an interface, so this approach won't work with IPsec Tunnel Mode.  IPsec Transport mode,
+  proposed in [IPSec transport mode support](https://github.com/submariner-io/enhancements/pull/103) will be required for
+  IPsec encryption.
+
+## TODO
+
+* Performance testing on POC code.
+* Support non-kubeproxy route agent handlers.
+* Support Loadbalancer mode.
+* Support detection and handling of gateway failure and recovery.
+
+## Links
+
+* Source for images:
+  [Submariner-MAG-Drawings](https://docs.google.com/presentation/d/19W-0YJB0wDiEWDhs7MoLFgKEm5GkGs2g47jXMUnIrv0/edit#)
```

Comment:
```suggestion
  [Submariner-MAG-Drawings](https://docs.google.com/presentation/d/1PsxbBbzt9A9opO4bVxkJcLnzD6zAQTlKtK2_aq5SINs/edit#)
```
The previous link was the "Axon Data Plane Overview" slides. I added the link before I knew which we were going to use.